instrumentation/opentelemetry/init.go

```diff
@@ -101,6 +114,35 @@ func makeExporterFactory(cfg *config.AgentConfig) func() (sdktrace.SpanExporter,
 	}
 }
 
+func createTLSConfig(reportingCfg *config.Reporting) *tls.Config {
+	tlsConfig := &tls.Config{}
+	tlsConfig.InsecureSkipVerify = !reportingCfg.GetSecure().GetValue()
+	certFile := reportingCfg.GetCertFile().GetValue()
+	if len(certFile) > 0 {
+		tlsConfig.RootCAs = createCaCertPoolFromFile(certFile)
+	}
+
+	return tlsConfig
+}
+
+// createCaCertPoolFromFile creates a CA Cert Pool from a file path containing
+// a raw CA certificate to verify a server certificate. The file path is the
+// reporting.cert_file config value.
+func createCaCertPoolFromFile(certFile string) *x509.CertPool {
```

Comment:
Shall we cover this function with tests?

Reply:
It's already covered by the unit testing the caller :). Just the negative scenarios are not covered

Reply:
done